### ISSUE TITLE
Implement IUserPhoneNumberStore?

### ISSUE THREAD

**Issue by sfmskywalker** [OrchardCMS/OrchardCore#5729]
I'm implementing an Orchard Core application that requires the user to enter & confirm their cell phone number, similar to the way email verification works. The latter is easy since `UserStore` implements `IUserEmailStore`, but when using `UserManager.GenerateChangePhoneNumberTokenAsync`, the call fails with a `NotSupportedException: Store does not implement IUserPhoneNumberStore<TUser>` exception.

The easiest & cleanest way I think would be for me would to provide a PR where `UserStore` implements `IUserPhoneNumberStore`, updating `User` accordingly by adding a `Phone` property.

Would there be any objections to doing so? 

@sebastienros @kevinchalet

**Comment by sfmskywalker** [comment]
FYI, for the time being, I'm registering a custom `UserStore` derivative that implements `IUserPhoneNumberStore<IUser>`, so I'm not blocked. But I think it might be beneficial if I implemented this in the User module directly.

**Comment by Skrypt** [comment]
Makes sense. Though how you want to handle the phone number(s) themselves on the IUser entity? I'm thinking that this would require to be a List<PhoneNumber> which would probably need a title/description "home, cellphone, work".

**Comment by sfmskywalker** [comment]
I was thinking along the lines of how the Email property is implemented on a user, of which there is only one. Same would go for Phone - just one property. This seems to be in line with the intended design of `IUserPhoneNumberStore<TUser>` in `Microsoft.AspNetCore.Identity`.

I would consider the ability of managing more than one email and phone numbers out of scope.

**Comment by Skrypt** [comment]
Ok makes sense after reading this : https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.iuserphonenumberstore-1?view=aspnetcore-3.1

**Comment by MikeAlhayek** [comment]
This exception is not longer an issue since we no longer support phone number. However, we should bring it back by providing full implementation as explained in #13710
